project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,365 @@
+//   * 
+//   * Criteria 1: Isn't a Daily Insider message - breaks ML model
+//   * Criteria 2: Isn't an empty email
+//   * Criteria 3: Isn't a google chat message
+//   */
+// function isActualEmail(emailBodyValue, emailDateValue) {
+//     if (emailBodyValue.indexOf("Daily Insider") != -1) {
+//         return false;
+//     }
+//     if (emailDateValue.length <= 1) {
+//         return false;
+//     }
+//     if (emailBodyValue.indexOf("format=flowed") != -1) {
+//         return false;
+//     }
+//     return true;
+// }
+
+// function getEmailObjects() {
+//     console.log(emailObjects);
+// }
+
+// /**
+//  * Get Message with given ID.
+//  *
+//  * @param  {String} messageId ID of Message to get.
+//  */
+// function getMessage(messageId) {
+//     const messageRequest = gapi.client.gmail.users.messages.get({
+//         'userId': 'me',
+//         'id': messageId,
+//         'format': "raw"
+//     });
+//     messageRequest.execute(response => {
+//         //Convert from base64 encoding to text.
+//         const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
+//         const emailBodyValue = getEmailBody(decodedEmail);
+//         const emailDateValue = getEmailDate(decodedEmail);
+//         const messageID = getMessageID(decodedEmail);
+//         if (isActualEmail(emailBodyValue, emailDateValue)) {
+//             emailObjects[messageID] = { emailDate: emailDateValue, emailBody: emailBodyValue };
+//         }
+//         getEmailObjects();
+//     });
+// }
+
+// //Retrieve messages using hardcoded queries and the signed-in email.
+// function listMessages() {
+//     emailObjects = {};
+//     var getPageOfMessages = function(request, result) {
+//         request.execute(function(resp) {
+//             result = result.concat(resp.messages);
+//             var nextPageToken = resp.nextPageToken;
+//             if (nextPageToken) {
+//                 request = gapi.client.gmail.users.messages.list({
+//                     'userId': 'me',
+//                     'pageToken': nextPageToken,
+//                     'q': getQuery()
+//                 });
+//                 getPageOfMessages(request, result);
+//             } else {
+//                 for (var i = 0; i < result.length; i++) {
+//                     if (result[i] === undefined) {
+//                         console.log("There are no messages from the query");
+//                     }
+//                     else {
+//                         console.log(result.length);
+//                         console.log(result);
+//                         this.getMessage(result[i].id);
+//                     }
+//                 }
+//             }
+//         });
+//     };
+//     var initialRequest = gapi.client.gmail.users.messages.list({
+//         'userId': 'me',
+//         'q': getQuery()
+//     });
+//     getPageOfMessages(initialRequest, []);
+// }
+
+class gmailAPI {
+    question;
```

Comment:
I think this can actually be removed and just set in the constructor as it already is.

Reply:
@brandonvu12 it works fine without me setting a type.

Reply:
@ederski does this mean that we review question and emailObjects here and just set it in the constructor as this.question or this.emailObjects, because when I tried that I got an error.

Reply:
I just did this! You were right!